### PR TITLE
Migrate version infos api

### PR DESCRIPTION
**Pull request by kritika-singh3**
Issue: #7482

Description: Migrating version_infos API from rails to spark keeping the version as v1

 - one difference - earlier if the user was not valid it use to return 404. Now it returns 403.

### CODE REVIEW

**Comment by kritika-singh3** [comment]
PR build: https://build.gocd.org/go/materials/value_stream_map/98e90b57b9b4cdbb974f8c7ddc7141e00fe6b617cc9ddf6f116394a410b7747e/01b0f0baff7ef0df90c483245208bd520e7b3607

File: api/api-version-infos-v1/src/main/java/com/thoughtworks/go/apiv1/versioninfos/VersionInfosControllerV1.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        path(controllerBasePath(), () -> {
+            before("", mimeType, this::setContentType);
+            before("/*", mimeType, this::setContentType);
+
+            before("", mimeType, this.apiAuthenticationHelper::checkUserAnd403);
+            before("/*", mimeType, this.apiAuthenticationHelper::checkUserAnd403);
+
+            get(Routes.VersionInfos.STALE, this::stale);
+            get(Routes.VersionInfos.LATEST_VERSION, this::latestVersion);
+            patch(Routes.VersionInfos.GO_SERVER, this::updateGoServerInfo);
```
Comment (GaneshSPatil) [inline api/api-version-infos-v1/src/main/java/com/thoughtworks/go/apiv1/versioninfos/VersionInfosControllerV1.java:73]: need to specify `mimeType`. Not specifying `mimeType` will cause the API to work for all the API versions.

Example: 
change:  `get(Routes.VersionInfos.STALE, this::stale);` 
to            `get(Routes.VersionInfos.STALE, mimeType, this::stale);`

Comment (kritika-singh3) [reply api/api-version-infos-v1/src/main/java/com/thoughtworks/go/apiv1/versioninfos/VersionInfosControllerV1.java:73]: Done: https://github.com/gocd/gocd/pull/8359

**Comment by maheshp** [comment]
Verified on: GoCD Version: 20.6.0 (11985-40a634e15e9898d802aa265fdc67f71350902813)

**Pull request merged**